spec/models/user_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:name) }
+    it { is_expected.to validate_presence_of(:email) }
+    it { is_expected.to validate_uniqueness_of(:email) }
+    it { is_expected.to allow_value("lamiya@gmail.com").for(:email) }
+    it { is_expected.to_not allow_value("lamiya@").for(:email) }
+    it { is_expected.to_not allow_value("lamiygmail.@com").for(:email) }
+    it { is_expected.to validate_presence_of(:role) }
+    it { is_expected.to define_enum_for(:role).with_values(%i[admin teacher student]) }
+  end
+
+  describe "associations" do
```

Comment:
```suggestion
  describe 'associations' do
```
Single quotes everywhere please